packages/dds/merge-tree/src/test/obliterate.reconnect.spec.ts

```diff
@@ -0,0 +1,162 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { IMergeTreeDeltaOp } from "../ops";
+import { TestClient } from "./testClient";
+import { TestClientLogger } from "./testClientLogger";
+
+type ClientName = "A" | "B" | "C" | "D";
+
+class ReconnectTestHelper {
+    A = new TestClient();
+    B = new TestClient();
+    C = new TestClient();
+    D = new TestClient();
+
+    get allClients() {
+        return [this.A, this.B, this.C, this.D];
+    }
+
+    idxFromName(name: ClientName): number {
+        return name.charCodeAt(0) - "A".charCodeAt(0);
+    }
+
+    logger = new TestClientLogger(this.allClients);
+
+    ops: ISequencedDocumentMessage[] = [];
+    perClientOps: ISequencedDocumentMessage[][] = this.allClients.map(() => []);
+
+    seq: number = 0;
+
+    constructor() {
+        this.A.startOrUpdateCollaboration("A");
```

Comment:
I think you might be able to simplify some of this helper and make it more flexible by re-using `createClientsAtInitialState`.